src/material/core/datetime/native-date-adapter.spec.ts

```diff
@@ -349,6 +349,12 @@ describe('NativeDateAdapter', () => {
     expect(adapter.isDateInstance(d)).toBe(false);
   });
 
+  it('should provide a method to return a valid date or null', () => {
+    let d = new Date();
+    expect(adapter.getValidDateOrNull(d)).toBe(d);
+    expect(adapter.getValidDateOrNull(new Date(NaN))).toBeNull();
```

Comment:
I would probably also add cases for `null`, `undefined`, `''`, `0`, a date string, and a timestamp just to be comprehensive.